### PR TITLE
(#937) New Feature: Teleport

### PR DESCRIPTION
**Pull request by AppleC4T**
- [x] Read and understood [see CONTRIBUTING.md](https://github.com/fossasia/labyrinth/blob/master/CONTRIBUTING.md)
- [x] Included a preview link and screenshot showing after and before the changes.
- [x] Included a description of change below.
- [x] Tried to [squash the commits][squash].
- [x] Added Screenshot / Path to tile to Indicate Location of Change in Game

### Changes done in this Pull Request

- View my work at https://pr-958-fossasia-labyrinth.surge.sh

- Fixes https://github.com/fossasia/labyrinth/issues/937

### The problem I want to solve or the facility I want to improve is/are

Add a teleport feature, enabling faster debugging/checking new tiles (As requested and described in https://github.com/fossasia/labyrinth/issues/937)

### My steps to solve it

Added a teleport feature.
 - The user can now type in a path e.g `UP,UP,DOWN`* and press a button to instantly move through typed path (the path takes a form of `DIRECTION,DIRECTION,DIRECTION...` the directions must be in capital letters and separated only with a comma as added in the path tracker)
 - The moves made via the teleport command do not count to the _moved times_ value, but this can be changed if requested.

### Screenshots, Location of Changes, Path to Tile

image
image
image

[squash]: https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git

### CODE REVIEW

**Comment by AppleC4T** [comment]
@YashKumarVerma check now. Everything should be ok now. 
 - added swal prompts for successful TPs, unsuccessful TPs and for invalid paths
 - input value resets after each TP command
 - the script now checks whether you can enter the tile
 - fixed the double quote error

File: js/player.js [hunk 1]
```diff
@@ -23,6 +23,61 @@ function createPlayerPicture() {
   return embed;
 }
 
+function runTp(){
+	console.log("tp");
```
Comment (YashKumarVerma) [inline js/player.js]: can we change it to something like **_teleporting / starting teleport / ..._**

File: js/player.js [hunk 2]
```diff
@@ -23,6 +23,61 @@ function createPlayerPicture() {
...
+	var isPathValid = true;
+	var isTpValid = true;
+	var TpPath = document.querySelector(".pathIn").value.split(",");
+	if(TpPath === []){
+		swal("Please provide a path.");
```
Comment (YashKumarVerma) [inline js/player.js]: we can use the swal error prompt.
you can pass 3 param in the same function. one with title. one with type. one with this message. 

refer their documentation.

File: js/player.js [hunk 3]
```diff
@@ -23,6 +23,61 @@ function createPlayerPicture() {
...
+					isTpValid = false;
+				}
+				break;
+			default:
+				isPathValid = false;
+		}
+	
+	}
+	if (isPathValid === false){
+		swal("Please provide a valid path (eg. UP,LEFT,DOWN,RIGHT)");
```
Comment (YashKumarVerma) [inline js/player.js]: same here. add error type to function as in line 32

File: js/player.js [hunk 4]
```diff
@@ -23,6 +23,61 @@ function createPlayerPicture() {
...
+		document.querySelector(".pathIn").value = "";
+	}else if (isTpValid === false) {
+		swal("Teleport unsuccessful. One of the tile can't be accessed from this direction.");
```
Comment (YashKumarVerma) [inline js/player.js]: add type of function in swal. (Line 32,70 need same)

File: js/player.js [hunk 5]
```diff
@@ -23,6 +23,61 @@ function createPlayerPicture() {
...
+		document.querySelector(".pathIn").value = "";
+	}else{
+		swal("Teleport successful!");
```
Comment (YashKumarVerma) [inline js/player.js]: add a success swal.

File: js/player.js (hunk 1 above)
Comment (YashKumarVerma) [reply js/player.js]: also, please use a descriptive name for function.

**Comment by AppleC4T** [comment]
@YashKumarVerma Ok, I'll get to it as soon as I can. 👍

**Comment by AppleC4T** [comment]
@YashKumarVerma Done.

File: js/player.js [hunk 6]
```diff
@@ -23,6 +23,64 @@ function createPlayerPicture() {
   return embed;
 }
 
+/*global runTeleport player:true*/
+/*eslint no-undef: "error"*/
```
Comment (YashKumarVerma) [inline js/player.js]: add swal here too

**Comment by AppleC4T** [comment]
Done.

**Comment by AppleC4T** [comment]
Thanks! 😄

**Comment by AppleC4T** [comment]
> p.s. GCI honors quality over quantity. so don't worry if it takes time.
Don't worry, I know. 😉

**Comment by AppleC4T** [comment]
@YashKumarVerma Done. The respective errors for no path and invalid path now trigger.

**Pull request merged**